airflow/sensors/sql_sensor.py

```diff
@@ -103,5 +103,5 @@ def poke(self, context):
             else:
                 raise AirflowException("self.success is present, but not callable -> {}".format(self.success))
         if self.allow_null:
-            return str(first_cell) not in ('0', '')
-        return str(first_cell) not in ('0', '', 'None')
+            return not (first_cell is not None or str(first_cell) == '' or int(first_cell) == 0)
```

Comment:
I'm not a fan of double-negatives. I think this is easier to understand without them. I think this works and carries the same meaning?

```suggestion
            return first_cell is None or first_cell != '' or first_cell != 0
```

We don't want

Reply:
Agree. I've found even more pythonic way of checking.